studio/studio.js

```diff
@@ -186,4 +282,32 @@ app.component('Tree', Tree)
 app.component('Menu', Menu)
 app.mount('#file-tree')
 document.onclick = e => eventBus.emit('documentClick', e)
-eventBus.emit("folder", $folder)
\ No newline at end of file
+eventBus.emit("folder", $folder)
+
+const handleHorizontalResize = (element, resizer) => {
+  let mouseX, elementWidth = 0
+  const
+    mouseMoveHandler = (e) => {
+      const dx = e.clientX - mouseX
+      if (elementWidth + dx > 60) element.style.width = elementWidth + dx + 'px'
+    },
+    mouseUpHandler = () => {
+      document.removeEventListener('mousemove', mouseMoveHandler)
+      document.removeEventListener('mouseup', mouseUpHandler)
+    },
+    mouseDownHandler = (e) => {
+      mouseX = e.clientX
+      const styles = window.getComputedStyle(element)
+      elementWidth = parseInt(styles.width, 10)
+      document.addEventListener('mousemove', mouseMoveHandler)
+      document.addEventListener('mouseup', mouseUpHandler)
+    }
+
+  resizer.addEventListener('mousedown', mouseDownHandler)
+}
+
+const
+  explorer = document.getElementById('explorer'),
+  widthResizer = document.getElementById('width-resizer')
```

Comment:
My intention was to make this reusable when we decide to implement the same behavior for code, console and preview views. But we can do it also later on if it will be applicable.

Reply:
We can always refactor when needed. The general rule of thumb is to make the code optimized for the current needs, but flexible enough for future extensions.